File: archive/factorial.c
#include <stdio.h>

long factorial(int n) {
  if (n < 2) {
    return 1;
  }

  return n * factorial(n - 1);
}

int main() {
  int number = 0;
  printf("Enter a number to find the factorial of: ");
  scanf("%d", &number);
  printf("%d\n", factorial(number));

  return 0;
}

File: archive/Factorial.pas
program Factorial;

function fact(const n: integer) : integer;
begin
    if (n = 0) then
        fact := 1
    else
        fact := n * fact(n - 1);
end;

var
  n: integer;

begin
  readln(n);
  writeln('Enter a number to find the factorial of:' );
  writeln(fact(n))
end.
